src/components/NewComment.jsx

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState } from "react";
+import { postComment } from "../utils/axiosFunctions";
+import CommentMessage from "./CommentMessage";
+
+export default function NewComment({
+  article_id,
+  username,
+  setCurrentComments,
+}) {
+  const [newComment, setNewComment] = useState(null);
+  const [tempComment, setTempComment] = useState("");
+  const [formSubmitted, setFormSubmitted] = useState(false);
+  const [commentMessage, setCommentMessage] = useState(null);
+  const [lockButton, setLockButton] = useState(false);
+
+  useEffect(() => {
+    if (newComment !== null && formSubmitted) {
+      postComment(article_id, { username: username, body: newComment }).then(
+        (postedComment) => {
+          setCommentMessage("success");
+          setLockButton(false);
+          setCurrentComments((currentComments) => {
+            const updatedComments = [postedComment, ...currentComments];
+            return updatedComments;
+          });
+        }
+      );
+    }
+    setNewComment(null);
```

Comment:
It's best to use one consistent datatype in states, so I'd suggest using an empty string `""` rather than `null`